File: src/main/java/io/github/docs/app/TransactionDocumentMetadata.java
package io.github.docs.app;

import io.github.docs.service.dto.SchemeDTO;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.HashSet;
import java.util.Set;

/**
 * This object contains the core metadata elements to display at the front end to represent
 * <p>
 * data about an attached document if only to be able to select a document without necessarily
 * <p>
 * reading it first, and further share by means like reassignment or  email
 */
@Data
@NoArgsConstructor
public class TransactionDocumentMetadata {


    private Long id;

    @NotNull
    private String transactionNumber;

    @NotNull
    private LocalDate transactionDate;

    private String briefDescription;

    private String justification;

    private BigDecimal transactionAmount;

    private String payeeName;

    private String invoiceNumber;

    private String lpoNumber;

    private String debitNoteNumber;

    private String logisticReferenceNumber;

    private String memoNumber;

    private String documentStandardNumber;

    private String transactionAttachmentContentType;

    @NotNull
    private String filename;

    private Set<SchemeDTO> schemes = new HashSet<>();

    @Override
    public String toString() {
        final StringBuilder sb = new StringBuilder("TransactionDocumentMetadataResource{");
        sb.append("transactionNumber='").append(transactionNumber).append('\'');
        sb.append(", transactionDate=").append(transactionDate);
        sb.append(", briefDescription='").append(briefDescription).append('\'');
        sb.append('}');
        return sb.toString();
    }
}

File: src/main/webapp/app/bespoke/model/transaction-document-metadata.ts
import { Moment } from 'moment';
import { IScheme } from 'app/shared/model/scheme.model';

export interface ITransactionDocumentMetadata {
  id?: number;
  transactionNumber?: string;
  transactionDate?: Moment;
  briefDescription?: string;
  justification?: string;
  transactionAmount?: number;
  payeeName?: string;
  invoiceNumber?: string;
  lpoNumber?: string;
  debitNoteNumber?: string;
  logisticReferenceNumber?: string;
  memoNumber?: string;
  documentStandardNumber?: string;
  filename?: string;
  schemes?: IScheme[];
  checked?: boolean;
}

export class TransactionDocumentMetadata implements ITransactionDocumentMetadata {
  constructor(
    public id?: number,
    public transactionNumber?: string,
    public transactionDate?: Moment,
    public briefDescription?: string,
    public justification?: string,
    public transactionAmount?: number,
    public payeeName?: string,
    public invoiceNumber?: string,
    public lpoNumber?: string,
    public debitNoteNumber?: string,
    public logisticReferenceNumber?: string,
    public memoNumber?: string,
    public documentStandardNumber?: string,
    public filename?: string,
    public schemes?: IScheme[],
    public checked?: boolean
  ) {}
}
